torch/csrc/distributed/rpc/tensorpipe_utils.cpp

```diff
@@ -163,12 +163,16 @@ std::tuple<tensorpipe::Message, TensorpipeWriteBuffers> tensorpipeSerialize(
     buffers.tensors = cloneSparseTensors(rpcMessage->tensors()).vec();
   }
 
+  int non_meta_idx = 0;
+  int meta_idx = std::count_if(buffers.tensors.begin(), buffers.tensors.end(), [](auto& t) { return !t.is_meta(); });
   torch::jit::Pickler pickler([&](const void* buf, size_t sz) -> size_t {
     buffers.pickle.insert(
         buffers.pickle.end(),
         static_cast<const char*>(buf),
         static_cast<const char*>(buf) + sz);
     return sz;
+  }, nullptr, nullptr, nullptr, [&](const at::Tensor& t) -> std::string {
+    return std::to_string(!t.is_meta() ? non_meta_idx++ : meta_idx++);
```

Comment:
this seems a bit finely balanced but I'm not sure about a good alternative. It seems like it would be better if the pickler natively understood meta tensors, so that you don't have to stub them out here.

Reply:
I was thinking about this, but it will require pickler/unpickler to track how many meta/non-meta tensors serialized so far and I decided that it is beyond the scope of this PR

Reply:
+1 This indeed looks a bit hacky. Shall we create an issue to track this (i.e., adding Meta tensor support to jit pickler/unpickler)? I assume we can treat it as if it is a zero-sized tensor?